src/foam/nanos/auth/AuthorizationException.java

```diff
@@ -5,15 +5,32 @@
  */
 
 package foam.nanos.auth;
+import foam.box.CapabilityRequiredRemoteException;
 
 /**
  * Throw this when a user is logged in, but is not permitted to do something.
  */
 public class AuthorizationException extends SecurityException {
+  // TODO: This should be decoupled from AuthorizationException after
+  //       the modelled exception feature is merged.
+  protected String permission_;
+
   public AuthorizationException() {
     super("Permission denied.");
   }
+  public AuthorizationException(String message, String permission) {
+    super(message + " (permission: '" + permission + "')");
```

Comment:
I don't think we should include the permission in the message since the end user will be seeing the message sometimes.